Add minutes (#66)
sardana-org/sardana-followup: 20200821/MINUTES.md

- PR (#1386) for adding scan widgets was sent.
- PR (#1355) for adding custom data to the scan was sent.
- added the possibility of adding custom data to the scan files (using
  the function addCustomData), the way used by spec was looked at.

MaXIV:
- they are testing python3 with the last release, they found some problems with
  python but the main problem was with nexus5. They manages to solve the
  problems using singularity and dockers and now they are ready. They have
  still problems with controllers talking directly to the hardware.
  Next week they want to deploy sardana3 for one beamline.
  The main issue is with controllers and macros developed by the users.
  David is working in two new controllers talking directly to the
  hardware. Some problems trying to understand the behaviour of the
  TriggerGateController, it does not work with the time scan.

Solaris:
- issues about Sardana customization. It was asked for an easy way to set
  the logging level (they want to set it to INFO) and about naming templates
  (the default ones do not follow their naming convention and renaming by hand
  is a lot of work). Zibi suggested to use the Sardana custom

>   settings. In solaris they use a sardana config file and also define there

Reviewer: I think it was Aureo who mentioned that they use `dsconfig` to easily rename Sardana elements.

  the name convention. 
- QtSpock: is integrated in a gui but once the scan is finished the command
  prompt does not appear, they found a way to workaround this but is not
  practicable for the users.
- they are now working on making scans depending on the status of the beam.

>   There are many implementations (DESY, MaXIV, Alb), it should be unified

Reviewer: ```suggestion
  There are many implementations (DESY, MaXIV, Alba), it should be unified
```

  (a dedicated meeting will be scheduled for that)
- how to import modules from controllers is an issue. SEP19 improves
  that but it is known that it should be improved in sardana. For now the
  python path environment could be used, or the PoolPath property of the Pool
  device.

DESY:

> - tests of sardana3 were done: it is already possibleto use at DESY with

Reviewer: ```suggestion
- tests of sardana3 were done: it is already possible to use at DESY with
```

  all the DESY related Sardana tools. Some beamlines use it.
- problems with fscan with virtual motors, an issue will be created if needed.

Extra issues reported by Zibi:
  - from Daniel Schick
    * asked for inverting the flow information between Sardana core and the
    controllers in case the controllers could push the information to the
    core (issue #1343). The compatibility with the current implementation
    should be kept.
    * asked for handling motor limits from the hardware controllers
    (issue #1344).
    Both issues require major changes. They will be investigated.
  - from David (MaXIV)
    * asked for timestamps in the scan, there is an issue about that. Perhaps
    could be configurable to get an extra timestamp for each channel, other
    question is how to put it in NeXus.

  - from Solaris
    * plugin interface for storing scan state, considering the case if the

>     scan was crashed and how to resume when starging again, it is not possible

Reviewer: ```suggestion
    scan was crashed and how to resume when starting again, it is not possible
```

    now. Alba also wanted to resume sequencies if they crash.
    - plugin interface for storing macroserver environment, for ex. Tango
    DB. With a plugin mechanism could be choosen where to store it.

2. Review pending points from the previous meeting
    - [x] Add migration guide for Sardana 2 -> Sardana 3 to
          the [docs](https://sardana-controls.org/devel/guide_migration/index.html)
	  It is done. 
 
    - [ ] Alba propose examples on how to program without `on_stop()` and `on_abort()` Macro methods.
          MAX IV and DESY need to test and review the new implementation.
	  Still not proposed by Alba, they will provide it.

    - [ ] MAX IV will provide a docker image to add to CI.
          It was some idea for extending the continuous integration, but is

> 	  still not done. We will have a look to was Taurus did, they use

Reviewer: ```suggestion
	  still not done. We will have a look to what Taurus did, they use
```
